runtime/src/bank.rs

```diff
@@ -2583,6 +2584,64 @@ impl Bank {
         debug!("Added native program {} under {:?}", name, program_id);
     }
 
+    /// Add a precompiled program
+    pub fn add_precompiled_program(&mut self, program_id: &Pubkey) {
+        // Adding a precompiled programs means ensuing that an executable
+        // accounts exists at its address
+        debug!("Adding precompiled program {}", program_id);
+        let existing_genuine_program =
+            if let Some(mut account) = self.get_account_with_fixed_root(program_id) {
+                // The account just needs to be executable
+                if account.executable() {
+                    Some(account)
+                } else {
+                    // malicious account is pre-occupying at program_id
+                    // forcibly burn and purge it
+
+                    self.capitalization.fetch_sub(account.lamports(), Relaxed);
+
+                    // Resetting account balance to 0 is needed to really purge from AccountsDb and
+                    // flush the Stakes cache
+                    account.set_lamports(0);
+                    self.store_account(program_id, &account);
+                    None
+                }
+            } else {
+                None
+            };
+
+        // introducing precompiled program
+
+        match &existing_genuine_program {
+            None => (), // continue to add account
+            Some(_account) => {
+                // nop; it seems that we already have account
+                return;
+            }
+        }
+
+        assert!(
+            !self.freeze_started(),
+            "Can't change frozen bank by adding not-existing new precompiled program ({}). \
+                Maybe, inconsistent program activation is detected on snapshot restore?",
+            program_id
+        );
+
+        // Add a bogus executable account, which will be loaded and ignored.
+        let (lamports, rent_epoch) =
+            self.inherit_specially_retained_account_fields(&existing_genuine_program);
+        let account = AccountSharedData::from(Account {
+            lamports,
+            owner: solana_sdk::system_program::id(),
```

Comment:
how about being consistent with a precompile program existing already on mainnet-beta?:

```
a$ ./target/release/solana -um account KeccakSecp256k11111111111111111111111111111

Public Key: KeccakSecp256k11111111111111111111111111111
Balance: 0.000000001 SOL
Owner: NativeLoader1111111111111111111111111111111
Executable: true
Rent Epoch: 0
Length: 17 (0x11) bytes
0000:   73 65 63 70  32 35 36 6b  31 5f 70 72  6f 67 72 61   secp256k1_progra
0010:   6d                                       
```

I mean, owner should be changed to `native_loader::id()`?

Reply:
Since the `KeccakSecp256k11111111111111111111111111111` account already exists on mainnet beta so  it will not be recreated, and therefore remain owned by the native loader.  Once the feature is activated it won't matter because precompiles only need an account that is executable